File: CVBuilder.Core/Models/UserModel.cs
namespace CVBuilder.Core.Models
{
    public class UserModel
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string PasswordHash { get; set; }

        public string? ResetToken { get; set; }
        public DateTime? ResetTokenExpires { get; set; }

        public List<CVModel>? CVs { get; set; }
    }
}

File: CVBuilder.Web/src/app/models/user.model.ts
import type { Cv } from './cv.model';

export interface User {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  passwordHash: string;

  resetToken?: string | null;
  /** ISO datetime string */
  resetTokenExpires?: string | null;

  cvs?: Cv[];
}
